src/main/java/org/folio/service/kbcredentials/KbCredentialsServiceImpl.java

```diff
@@ -95,15 +100,17 @@ public KbCredentialsServiceImpl(Converter<DbKbCredentials, KbCredentials> creden
     postBodyValidator.validate(entity);
     KbCredentials kbCredentials = entity.getData();
     final String tenantId = tenantId(okapiHeaders);
-    return verifyCredentials(kbCredentials, okapiHeaders)
-      .thenCompose(o -> fetchUserInfo(okapiHeaders))
+    return fetchUserInfo(okapiHeaders)
```

Comment:
code blocks in `save`, `update`, `updatePartially` looks very similar to me: consider refactoring them